classes/class-wc-connect-shipping-label.php

```diff
@@ -0,0 +1,310 @@
+			$formData[ 'cart' ] = $contents;
+
+			foreach( $order->get_address( 'shipping' ) as $key => $value ) {
+				$formData[ 'orig_' . $key ] = '';
+				$formData[ 'dest_' . $key ] = $value;
+			}
+
+			return $formData;
+		}
+
+		protected function get_form_layout() {
+			$layout = [];
+
+			$address_fields = array(
+				array(
+					'key' => 'first_name',
+				),
+				array(
+					'key' => 'last_name',
+				),
+				array(
+					'key' => 'company',
+				),
+				array(
+					'key' => 'address_1',
+				),
+				array(
+					'key' => 'address_2',
+				),
+				array(
+					'key' => 'city',
+				),
+				array(
+					'key' => 'state',
+				),
+				array(
+					'key' => 'postcode',
+				),
+				array(
+					'key' => 'country',
+				),
+			);
+			$address_summary = '{first_name} {last_name}\\n{address_1} {address_2}\\n{city}, {postcode} {state}, {country}';
+
+			$orig_address_fields = $address_fields;
+			$dest_address_fields = $address_fields;
+			foreach( $address_fields as $index => $_ ) {
+				$orig_address_fields[ $index ][ 'key' ] = 'orig_' . $address_fields[ $index ][ 'key' ];
+				$dest_address_fields[ $index ][ 'key' ] = 'dest_' . $address_fields[ $index ][ 'key' ];
+			}
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Origin',
+				'title' => 'Generate shipping label: Address verification (origin)',
+				'description' => 'asdfghjkjhgfdsaASDFGHJJHGFDSA',
+				'items' => array( array( 'items' => $orig_address_fields ) ),
+				'bypass_suggestion_flag' => 'orig_bypass_suggestion',
+				'summary' => str_replace( '{', '{orig_', $address_summary ),
+				'sugggestion_hint' => 'vnkdjfitysncgkgse,jhxccvnaluerghsldjfvbzdj',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Destination',
+				'title' => 'Generate shipping label: Address verification',
+				'description' => 'asdfghjkjhgfdsaASDFGHJJHGFDSA',
+				'items' => array( array( 'items' => $dest_address_fields ) ),
+				'bypass_suggestion_flag' => 'dest_bypass_suggestion',
+				'summary' => str_replace( '{', '{dest_', $address_summary ),
+				'sugggestion_hint' => 'qwertyuiopqwertyuiopqwerty uiopqwertyuiopqwertyuiop',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Packages',
+				'title' => 'Shopping cart contents',
+				'items' => array(
+					array(
+						'items' => array(
+							array(
+								'key' => 'cart',
+								'type' => 'shopping_cart',
+							),
+						),
+					),
+				),
+				'summary' => '{cart}',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Rates',
+				'title' => 'Select a shipping service',
+				'items' => array(
+					array(
+						'items' => array(
+							array(
+								'key' => 'rate',
+								'type' => 'radios',
+								'titleMap' => array(
+									'media' => 'Media Mail ($1.00)',
+									'pri_1' => 'Priority 1-day ($9.99)',
+								),
+							),
+						),
+					),
+				),
+				'summary' => '{rate}',
+			);
+
+			$layout[] = array(
+				'type' => 'summary',
```

Comment:
This is a special type of step, which will need to have access to the whole schema / layout / data. It will show the summary of every past step. There are no mock-ups of this step yet, but I assume something like this will be needed.